backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -365,4 +418,362 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
         .thenCompose(unused -> artifactHandler.deleteArtifacts(runId, optionalKeys), executor)
         .thenCompose(unused -> updateModifiedTimestamp(runId, now), executor);
   }
+
+  private InternalFuture<UserInfo> getCurrentLoginUserInfo() {
+    return FutureGrpc.ClientRequest(
+        uac.getUACService().getCurrentUser(Empty.newBuilder().build()), executor);
+  }
+
+  public InternalFuture<ExperimentRun> createExperimentRun(final CreateExperimentRun request) {
+    // Validate arguments
+    var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              String errorMessage = null;
+              if (request.getProjectId().isEmpty() && request.getExperimentId().isEmpty()) {
+                errorMessage =
+                    "Project ID and Experiment ID not found in CreateExperimentRun request";
+              } else if (request.getProjectId().isEmpty()) {
+                errorMessage = "Project ID not found in CreateExperimentRun request";
+              } else if (request.getExperimentId().isEmpty()) {
+                errorMessage = "Experiment ID not found in CreateExperimentRun request";
+              }
+
+              if (errorMessage != null) {
+                throw new InvalidArgumentException(errorMessage);
+              }
+            },
+            executor);
+
+    return futureTask
+        .thenCompose(
+            unused ->
+                checkProjectPermission(
+                    Collections.singletonList(request.getProjectId()),
+                    ModelDBActionEnum.ModelDBServiceActions.UPDATE),
```

Comment:
We need to check if the user can update the project and create the ER. We're missing the create part.